src/librustc_yk_sections/mir_cfg.rs

```diff
@@ -61,34 +61,32 @@ impl<'a, 'tcx, 'gcx> ConvCx<'a, 'tcx, 'gcx> {
         }
     }
 
-    /// Issues a fresh TIR variable index.
-    fn next_tir_var(&self) -> TirVarIndex {
+    /// Returns a guaranteed unique TIR variable index.
+    fn new_tir_var(&self) -> TirVarIndex {
         let var_idx = self.next_tir_var.get();
         self.next_tir_var.set(var_idx + 1);
         var_idx
     }
 
-    /// Get the TIR variable corresponding with the MIR variable, creating a fresh one if needed.
-    fn get_tir_var(&self, local: Local) -> TirVarIndex {
+    /// Get the TIR variable for the specified MIR variable, creating a fresh variable if needed.
+    fn tir_var(&self, local: Local) -> TirVarIndex {
         let local_u32 = local.as_u32();
         let mut var_map = self.var_map.borrow_mut();
 
         // Resize the backing Vec if necessary.
-        if var_map.len() <= usize::try_from(local_u32).unwrap() {
-            var_map.resize(usize::try_from(local_u32 + 1).unwrap(), None);
+        // Vector indices are `usize`, but variable indices are `u32`, so converting from a
+        // variable index to a vector index is always safe if a `usize` can express all `u32`s.
+        debug_assert!(size_of::<usize>() >= size_of::<u32>());
```

Comment:
This should be an `assert`. If it's true, the optimiser will remove it, but it means that even people on release mode will find out if (somehow) our code will break.

Reply:
Good point. Shall I change all of the numeric range debug assertions to full on assertions?

Reply:
Yep.